notebooks/features/cognitive_services/CognitiveServices - OpenAI.ipynb

```diff
@@ -488,6 +489,274 @@
     "\n",
     "display(completion.transform(qa_df))"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# OpenAI Embeddings\n",
+    "\n",
+    "We will use t-SNE to reduce the dimensionality of the embeddings from 1536 to 2. Once the embeddings are reduced to two dimensions, we can plot them in a 2D scatter plot."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from synapse.ml.cognitive import OpenAIEmbedding\n",
+    "\n",
+    "embedding = (\n",
+    "    OpenAIEmbedding()\n",
+    "    .setSubscriptionKey(key)\n",
+    "    .setDeploymentName(deployment_name_embeddings)\n",
+    "    .setUrl(\"https://{}.openai.azure.com/\".format(service_name))\n",
+    "    .setTextCol(\"combined\")\n",
+    "    .setErrorCol(\"error\")\n",
+    "    .setOutputCol(\"embeddings\")\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyspark.sql.functions as F\n",
+    "\n",
+    "df = spark.read.options(inferSchema=\"True\", delimiter=\",\", header=True).csv(\n",
+    "    \"wasbs://publicwasb@mmlspark.blob.core.windows.net/fine_food_reviews_1k.csv\"\n",
+    ")\n",
+    "\n",
+    "df = df.withColumn(\n",
+    "    \"combined\",\n",
+    "    F.format_string(\"Title: %s; Content: %s\", F.trim(df.Summary), F.trim(df.Text)),\n",
+    ")\n",
+    "\n",
+    "display(df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyspark.sql.functions import col\n",
+    "\n",
+    "completed_df = embedding.transform(df).cache()\n",
+    "display(completed_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieve embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "matrix = np.array(completed_df.select(\"embeddings\").collect())[:, 0, :]\n",
+    "matrix.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reduce dimensionality\n",
+    "We reduce the dimensionality to 2 dimensions using t-SNE decomposition."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from sklearn.manifold import TSNE\n",
+    "import numpy as np\n",
+    "\n",
+    "# Create a t-SNE model and transform the data\n",
+    "tsne = TSNE(\n",
+    "    n_components=2, perplexity=15, random_state=42, init=\"random\", learning_rate=200\n",
```

Comment:
nit, probabbly dont need to set any params other than n_components. It will perform better with defaults